### ISSUE TITLE
Using filter widget with drop box and input type text

### ISSUE THREAD

**Issue by artlemaks** [Mottie/tablesorter#1492]
Hi, 

I have looked through some of previous issues and tried to apply the suggested solutions but it didn't seem to work. 

I am using stickyHeader and filter widgets from the tablesorter. The filter worked fine until I changed my cells to allow for input. Now I can't seem to be able to search for anything.

I have tried a solution which suggested adding filter: "parsed" in the headers section but that doesn't seem to work. Here is jsFiddle of my code: http://jsfiddle.net/phcjb314/

**Comment by Mottie** [comment]
Hi @artlemaks!

I think what's missing is the `parser-input-select.js` file; you'll see this file in the external resources of the demo. When it's loaded in, it watches for changes within the table inputs & selects, and updates the internal cache automatically.

```html
<script src="js/parsers/parser-input-select.js"></script>
```

Alternatively, there is a [content editable widget](https://mottie.github.io/tablesorter/docs/example-widget-editable.html) which accomplishes the same thing without the appearance of an input element in the table cell.

**Comment by artlemaks** [comment]
Hi @Mottie ! 

Thank you for the reply. I have tried adding the suggested script but it still doesn't seem to work.

Here is what I currently have in the tablesorter code:

`$.tablesorter.addParser({
						id: 'input',
						is: function(s) {
							return false;
						},
						format: function(s, table, cell) {
							return $('input', cell).val();
						},
						type: 'numeric'
					});
					
					$.tablesorter.addParser({
						id: 'input2',
						is: function(s) {
							return false;
						},
						format: function(s, table, cell) {
							return $('input', cell).val();
						},
						type: 'text'
					});
					
					
					$("#tbl_store_product_sizes").tablesorter({
						widthFixed : true,
						showProcessing: true,
						headerTemplate : '{content} {icon}',
						headers: {
							// 0 is the index of the first column (zero-based)
							1: { sorter: 'input2', filter: "input2" },
							
							3: { sorter: 'input' },
							4: { sorter: 'input' },
							5: { sorter: 'input' }
						},
						 widgets: ['columns', 'stickyHeaders', 'filter'],
						  widgetOptions: {
							// jQuery selector or object to attach sticky header to
							stickyHeaders_attachTo: '#form_modal_body',
							stickyHeaders_offset: 0,
							stickyHeaders_cloneId : '-sticky',
							
							filter_cssFilter   : 'form-control input-sm',
							filter_childRows   : false,
							filter_hideFilters : false,
							filter_ignoreCase  : true,
							filter_useParsedData: true,
							filter_searchDelay : 30,
							filter_startsWith  : false,
							filter_functions : {
							}
						  }
					});`

When I then try to search for an item, I get the following results http://easycount.io/platformDev/searchDemo.png and one of the products is Oil Rapeseed (5Lt) so it should find matches. I'm probably missing something very simple in there..? 

I also realized I posted the wrong jsFiddle link, here is the proper one https://jsfiddle.net/x8e3mdmf/4/

**Comment by Mottie** [comment]
Within the `parser-input-select` file are different parsers. Most columns will need the `inputs` parser, and the select column will need the `select-text` parser since I'm guessing the select has numeric values.

Instead of using the `header` because it gets confusing, I removed the section and added the class names to the HTML ([demo](https://jsfiddle.net/Mottie/x8e3mdmf/7/)):

```html
<thead>
  <tr>
    <th>No.</th>
    <th class="sorter-inputs">Product Name</th>
    <th class="sorter-select-text">Department Name</th>
    <th class="sorter-inputs">Unit Cost</th>
    <th class="sorter-inputs">Quantity Counted</th>
    <th class="sorter-inputs">Total Cost</th>
  </tr>
</thead>
``` 

All I changed in the JS was removing the two `addParser` functions since they wouldn't be necessary.
